shub/settings.py

```diff
@@ -0,0 +1,537 @@
+
+# Password validation
+# https://docs.djangoproject.com/en/1.9/ref/settings/#auth-password-validators
+
+AUTH_PASSWORD_VALIDATORS = [
+    {
+        "NAME": "django.contrib.auth.password_validation.UserAttributeSimilarityValidator"
+    },
+    {"NAME": "django.contrib.auth.password_validation.MinimumLengthValidator"},
+    {"NAME": "django.contrib.auth.password_validation.CommonPasswordValidator"},
+    {"NAME": "django.contrib.auth.password_validation.NumericPasswordValidator"},
+]
+
+
+# Internationalization
+# https://docs.djangoproject.com/en/1.9/topics/i18n/
+
+LANGUAGE_CODE = "en-us"
+# TIME_ZONE = 'America/Chicago'
+USE_I18N = True
+USE_L10N = True
+USE_TZ = True
+
+SESSION_SERIALIZER = "django.contrib.sessions.serializers.PickleSerializer"
+SENDFILE_BACKEND = "sendfile.backends.development"
+PRIVATE_MEDIA_REDIRECT_HEADER = "X-Accel-Redirect"
+CRISPY_TEMPLATE_PACK = "bootstrap3"
+
+CACHES = {"default": {"BACKEND": "django.core.cache.backends.locmem.LocMemCache"}}
+
+# Static files (CSS, JavaScript, Images)
+# https://docs.djangoproject.com/en/1.9/howto/static-files/
+
+MEDIA_ROOT = "/var/www/images"
+MEDIA_URL = "/images/"
+STATIC_ROOT = "/var/www/static"
+STATIC_URL = "/static/"
+UPLOAD_PATH = "%s/_upload" % MEDIA_ROOT
+
+
+RQ_QUEUES = {"default": {"URL": os.getenv("REDIS_URL", "redis://redis/0")}}
+
+RQ = {"host": "redis", "db": 0}
+
+# background tasks
+BACKGROUND_TASK_RUN_ASYNC = True
+
+# Plugins
+
+# If PAM_AUTH in plugins enbled, add django_pam
+if "pam_auth" in PLUGINS_ENABLED:
+    INSTALLED_APPS += ["django_pam"]
+
+# If google_build in use, we are required to include GitHub
+if "google_build" in PLUGINS_ENABLED:
+
+    # For task discovery by celery
+    SOCIAL_AUTH_GITHUB_SCOPE = [
+        "admin:repo_hook",
+        "repo:status",
+        "user:email",
+        "read:org",
+        "admin:org_hook",
+        "deployment_status",
+    ]
+    ENABLE_GITHUB_AUTH = True
+
+# Apply any plugin settings
+for plugin in PLUGINS_ENABLED:
+
+    plugin_module = "shub.plugins." + plugin
+    plugin = import_module(plugin_module)
+
+    # Add the plugin to INSTALLED APPS
+    INSTALLED_APPS.append(plugin_module)
+
+    # Add AUTHENTICATION_BACKENDS if defined, for authentication plugins
+    if hasattr(plugin, "AUTHENTICATION_BACKENDS"):
+        AUTHENTICATION_BACKENDS = (
+            AUTHENTICATION_BACKENDS + plugin.AUTHENTICATION_BACKENDS
+        )
+
+    # Add custom context processors, if defines for plugin
+    if hasattr(plugin, "CONTEXT_PROCESSORS"):
+        for context_processor in plugin.CONTEXT_PROCESSORS:
+            TEMPLATES[0]["OPTIONS"]["context_processors"].append(context_processor)
+
+# Finally, ensure all variables in cfg are set in locals
+for key, value in cfg:
+    locals()[key] = value
+
+# Add SREGISTRY_SECRET_ from the environment
+for key, value in os.environ.items():
+    if key.startswith("SREGISTRY_SECRET_"):
+        secret_key = key.replace("SREGISTRY_SECRET_", "")
+        if not value:
```

Comment:
You want `if value is not None:` here, otherwise if `value` is "falsey" like `0` it will not be set.